File: src/main/java/io/github/support/terminal/application/domains/core/api/telegram/models/Message.java
package io.github.support.terminal.application.domains.core.api.telegram.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.util.Date;

@Data
@ToString
@Accessors(chain = true)
public class Message {

    @JsonProperty("message_id")
    private String messageId;

    @JsonProperty("from")
    private User from;

    @JsonProperty("chat")
    private Chat chat;

    @JsonProperty("date")
    private Date date;

    @JsonProperty("text")
    private String text;

}

File: src/main/kotlin/io/github/support/terminal/application/domains/integrations/telegram/models/Message.kt
package io.github.support.terminal.application.domains.integrations.telegram.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
data class Message(
        @JsonProperty("message_id") val messageId: String?,
        @JsonProperty("from") val from: User,
        @JsonProperty("chat") val chat: Chat,
        @JsonProperty("date") val date: Date,
        @JsonProperty("text") val text: String?
)